opentera-webrtc-native-client/OpenteraWebrtcNativeClient/python/package/CMakeLists.txt

```diff
@@ -1,41 +1,118 @@
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    VERBATIM
+)
+add_custom_target(
+    python-opentera-webrtc-native-client-so
+    ALL
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/so.md5
+    VERBATIM
+)
 
 # Generate html doc
 if (OPENTERA_WEBRTC_ENABLE_PYTHON_HTML_DOC)
-    add_custom_target(pydoc-html-file ALL
-	    COMMAND python3 -m pydoc -w opentera_webrtc_native_client
-	    WORKING_DIRECTORY ${LIBRARY_OUTPUT_PATH}
-	    DEPENDS opentera_webrtc_native_client)
+    add_custom_command(
+        OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/${PYTHON_LIB_NAME}.html
+        DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/so.md5 ${GENERATE_CONFIGURE_OUTPUT_FILES} python-opentera-webrtc-native-client-so
+        COMMAND /usr/bin/env python3 -m pydoc -w ${PYTHON_LIB_NAME}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+        VERBATIM
+    )
+    add_custom_target(
+        python-opentera-webrtc-native-client-html
+        ALL
+        DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/${PYTHON_LIB_NAME}.html
+        VERBATIM
+    )
+    set(HTML_DOC_DEPENDENCY_OPTION python-opentera-webrtc-native-client-html ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/${PYTHON_LIB_NAME}.html)
+else()
+    set(HTML_DOC_DEPENDENCY_OPTION ${CMAKE_CURRENT_BINARY_DIR}/so.md5 ${GENERATE_CONFIGURE_OUTPUT_FILES} python-opentera-webrtc-native-client-so)
+endif()
 
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    DEPENDS ${HTML_DOC_DEPENDENCY_OPTION}
+    COMMAND /usr/bin/env python3 setup.py bdist bdist_wheel sdist
+    COMMAND tar c ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/dist 2> /dev/null | md5sum > ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    VERBATIM
+)
+add_custom_target(
+    python-opentera-webrtc-native-client-wheel
+    ALL
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    VERBATIM
+)
 
-    install(FILES ${LIBRARY_OUTPUT_PATH}/opentera_webrtc_native_client.html DESTINATION python/package)
-endif()
+file(GENERATE
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/cmake/pip_install.cmake
+    CONTENT
+    "execute_process(
+        COMMAND /usr/bin/env python3 -m pip install --upgrade --force-reinstall --no-deps --no-cache-dir --no-index --find-links=${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/dist opentera-webrtc-native-client
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    )
+    execute_process(
+        COMMAND bash -c \"tar c $(/usr/bin/env python3 -m pip show opentera-webrtc-native-client | perl -n0777E 'say $1 if /Location:\\\\s+(.*?)\\\\n.*/s')/opentera_webrtc_native_client 2> /dev/null | md5sum > ${CMAKE_CURRENT_BINARY_DIR}/install.md5\"
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    )"
+)
```

Comment:
Use a cmake script for the installation step, to reuse the code both in the custom_command when installing on build, and the install() target when installing on install